cron/main.go

```diff
@@ -32,67 +35,48 @@ type Repository struct {
 }
 
 func main() {
-	fileName := fmt.Sprintf("%02d-%02d-%d.json", time.Now().Month(), time.Now().Day(), time.Now().Year())
-	result, err := os.Create(fileName)
-	if err != nil {
-		panic(err)
-	}
-	defer result.Close()
 	projects, err := os.OpenFile(os.Args[1], os.O_RDONLY, 0644)
 	if err != nil {
 		panic(err)
 	}
 	defer projects.Close()
 
-	repos := []Repository{}
-	data, err := ioutil.ReadAll(projects)
-	if err != nil {
+	inputRepos := []Repository{}
+	if data, err := ioutil.ReadAll(projects); err != nil {
 		panic(err)
-	}
-	err = csvutil.Unmarshal(data, &repos)
-	if err != nil {
+	} else if err = csvutil.Unmarshal(data, &inputRepos); err != nil {
 		panic(err)
 	}
-	sort.Slice(repos, func(i, j int) bool {
-		return repos[i].Repo < repos[j].Repo
-	})
-
-	//nolint
-	const checks string = "--checks=Active,CI-Tests,CII-Best-Practices,Code-Review,Contributors,Frozen-Deps,Fuzzing,Packaging,Pull-Requests,SAST,Security-Policy,Signed-Releases,Signed-Tags"
 
-	if _, err = result.Write([]byte(`{"results":[`)); err != nil {
+	fileName := fmt.Sprintf("%02d-%02d-%d.json",
+		time.Now().Month(), time.Now().Day(), time.Now().Year())
+	result, err := os.Create(fileName)
+	if err != nil {
 		panic(err)
 	}
-	for i, r := range repos {
+	for _, r := range inputRepos {
 		fmt.Println(r.Repo)
-		//nolint
-		cmd := exec.Command("../scorecard", fmt.Sprintf("--repo=%s", r.Repo), fmt.Sprintf("--metadata=%s", r.Metadata), checks, "--show-details", "--format=json")
-		// passing the external github token the cmd
-		cmd.Env = append(cmd.Env, fmt.Sprintf("GITHUB_AUTH_TOKEN=%s", os.Getenv("GITHUB_AUTH_TOKEN")))
-		cmd.Stderr = io.Writer(os.Stderr)
 
-		// Execute the command
-		data, err := cmd.Output()
-		if err != nil {
-			fmt.Printf("error:failed for the repo %s and the error is %s", r.Repo, err.Error())
-			// continuing because this is just for that repo that failed.
-			continue
+		repoUrl := repos.RepoURL{}
+		if err := repoUrl.Set(r.Repo); err != nil {
+			panic(err)
 		}
-
-		_, err = result.WriteString(string(data))
-		if err != nil {
+		if err := repoUrl.ValidGitHubUrl(); err != nil {
 			panic(err)
 		}
-		if i < len(repos)-1 {
-			_, err = result.WriteString(",")
-			if err != nil {
-				panic(err)
-			}
+
+		ctx := context.Background()
+		cfg := zap.NewProductionConfig()
+		cfg.Level.SetLevel(zap.InfoLevel)
+		logger, _ := cfg.Build()
+		sugar := logger.Sugar()
+		repoResult := pkg.RunScorecards(ctx, sugar, repoUrl, checks.AllChecks)
+		if err := repoResult.AsJSON( /*showDetails=*/ true, result); err != nil {
```

Comment:
(i'll start a thread with the relevant people).

Reply:
I saw the thread and looks like folk are ok with the breaking change? So leaving this as is for now.

FWIW, I think we should ask users to stop relying on these *.json files completely and instead use BQ. These files are internal IMO and shouldn't be exposed to users. Else, we will end up in a situation where any improvements to our architecture will require users to update their system.

Reply:
Yes, let's proceed with this.

We should have some intermediate format of some sort, but we may still want a stable GCS structure in the future of some sort, as it's easily usable compared to BQ (since GCS buckets can be accessed over HTTP).

Reply:
SG. Let's discuss more on the intermediate format on the doc and during our 1:1.